openpype/lib/transcoding.py

```diff
@@ -315,6 +315,92 @@ def parse_oiio_xml_output(xml_string, logger=None):
     return output
 
 
+def get_channels_info_by_layer_name(channel_names):
+    """Get channels info grouped by layer name.
+
+    Finds all layers in channel names and returns list of dictionaries with
+    information about channels in layer.
+    Example output (not real world example):
+        [
+            {
+                "name": "Main",
+                "review_channels": {
+                    "R": "Main.red",
+                    "G": "Main.green",
+                    "B": "Main.blue",
+                    "A": None,
+                },
+                "channel_names": {
+                    "Main.red": "red",
+                    "Main.green": "green",
+                    "Main.blue": "blue",
+                    "Main.z": "z"
+                }
+            },
+            {
+                "name": "Composed",
+                "review_channels": {
+                    "R": "Composed.R",
+                    "G": "Composed.G",
+                    "B": "Composed.B",
+                    "A": "Composed.A",
+                },
+                "channel_names": {
+                    "Composed.R": "R",
+                    "Composed.G": "G",
+                    "Composed.B": "B",
+                    "Composed.A": "A",
+                }
+            },
+            ...
+        ]
+
+    Args:
+        channel_names (list[str]): List of channel names.
+
+    Returns:
+        list[dict]: List of channels information.
+    """
+
+    layer_names_order = []
+    rgba_by_layer_name = collections.defaultdict(dict)
+    channels_by_layer_name = collections.defaultdict(dict)
+    for channel_name in channel_names:
+        name_parts = channel_name.split(".")
+        last_part = name_parts.pop(-1)
+        layer_name = ".".join(name_parts)
+        channels_by_layer_name[layer_name][channel_name] = last_part
+        rgb_part = last_part.lower()
+        if layer_name not in layer_names_order:
+            layer_names_order.append(layer_name)
+        if rgb_part in ("r", "red"):
+            rgba_by_layer_name[layer_name]["R"] = channel_name
+        elif rgb_part in ("g", "green"):
+            rgba_by_layer_name[layer_name]["G"] = channel_name
+        elif rgb_part in ("b", "blue"):
+            rgba_by_layer_name[layer_name]["B"] = channel_name
+        elif rgb_part in ("a", "alpha"):
+            rgba_by_layer_name[layer_name]["A"] = channel_name
```

Comment:
I wonder if something like this would make more sense:
```suggestion
        match = re.match(r"^(.*\).(r|red|g|green|b|blue|a|alpha)$", flags=re.IGNORECASE)
        if match:
            # ends with rgba identifier, e.g. main.red or main.A
            layer_name = match.group(0)
            rgb_part = match.group(1)
            rgb_channel = rgb_part[0].upper()  # R, G, B or A
            rgba_by_layer_name[layer_name][rgb_channel] = channel_name
        else:
            # do whatever if it does not end with rgba identifier
```

Reply:
Or just:
```suggestion
        if "." in channel_name:
            layer_name, last_part = channel_name.rsplit(".", 1)
        else:
            layer_name = ""
            last_part = channel_name

        channels_by_layer_name[layer_name][channel_name] = last_part
        if layer_name not in layer_names_order:
            layer_names_order.append(layer_name)

        if last_part.lower() in {"r", "red", "g", "green", "b", "blue", "a", "alpha"}:
            channel = last_part[0].upper() # R, G, B or A
            rgba_by_layer_name[layer_name][channel] = channel_name
```